File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/Animal.java
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass;

/**
 * Demonstrate the abstract class
 *
 * @author Ranjith Krishnamurthy
 */
public abstract class Animal {
    private String DNA;

    public Animal(String DNA) {
        this.DNA = DNA;
    }

    public abstract void makeNoise();

    public void displayDNA() {
        System.out.println(DNA);
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/Animal.kt
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass

/**
 * Demonstrate the abstract class
 *
 * @author Ranjith Krishnamurthy
 */
abstract class Animal(private val DNA: String) {
    abstract fun makeNoise()
    fun displayDNA() {
        println(DNA)
    }
}